src/components/layout/Search.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+
+const Search = ({ onSearch }) => {
+  const [input, setInput] = useState('');
+
+  const handleChange = (e) => {
+    setInput(e.target.value);
+  };
+
+  useEffect(() => {
+    if (input === '') {
+      onSearch(input);
+    }
+    // eslint-disable-next-line
+  }, [input]);
```

Comment:
Although this works, it's best practice to include all of the necessary dependencies in your dependency array. Since adding `onSearch` to the array will cause issues, it might be a better idea handle this behavior in the `clear` function.